src/ArcCommander/Configuration.fs

```diff
@@ -0,0 +1,208 @@
+    ///
+    /// Returns the values assigned to a given key across all sections
+    let getAllValuesOfKey key (configuration:IniData) =
+        configuration.Sections
+        |> Seq.choose (fun s ->
+            if s.Keys.ContainsKey key then
+                Some s.Keys.[key]
+            else None
+        )
+
+    /// Returns the value assigned to a specific name (section+key)
+    ///
+    /// The name is given as string in form "section.key"
+    let tryGetValueOfName (name:string) (configuration:IniData) =
+        try
+            let section,key =  splitName name 
+            tryGetSection section configuration
+            |> Option.bind (tryGetValue key)
+        with 
+        | err -> 
+            printfn "Error: Could not retrieve value with given name\n %s" err.Message
+            None
+
+    /// Returns true if the name (section+key) is set in the configuration
+    ///
+    /// The name is given as string in form "section.key"
+    let nameExists (name:string) (configuration:IniData) =
+        let section,key = splitName name 
+        tryGetSection section configuration
+        |> Option.bind (tryGetValue key)
+        |> Option.isSome
+
+    /// If the name is already set in the config, assigns a new value to it
+    ///
+    /// The name is given as string in form "section.key"
+    let setValue (name:string) (value:string) (configuration:IniData) =
+        if nameExists (name:string) (configuration:IniData) then
+            let section,key = splitName name 
+            configuration.[section].[key] <- value
+            configuration
+        else
+            printfn "name %s does not exist in the config" name
+            configuration
+
+    /// If the name is not already set in the config, adds it together with the given value
+    ///
+    /// The name is given as string in form "section.key"
+    let addValue (name:string) (value:string) (configuration:IniData) =
+        if nameExists (name:string) (configuration:IniData) then
+            printfn "name %s already exists in the config" name
+            configuration
+        else
+            let section,key = splitName name 
+            configuration.[section].AddKey(key,value) |> ignore
+            configuration
+
+    /// Merges the setting from two configurations. If a name is contained in both files, the value bound to this name in the localConfig is used
+    let merge (localConfig:IniData) (globalConfig:IniData) = 
+        globalConfig.Merge localConfig
+        globalConfig
+
+    /// Returns a collection of all name value pairs in the config
+    ///
+    /// The names are given as string in form "section.key"
+    let flatten (configuration:IniData) =
+        configuration.Sections
+        |> Seq.collect (fun s ->
+            s.Keys
+            |> Seq.map (fun kv -> s.SectionName+"."+kv.KeyName,kv.Value)
+        )
+
+    /// Gets the current configuration
+    let loadConfiguration workdir =
```

Comment:
this function does not do what the name suggests, maybe the name should be something like `getMergedConfigurations`? Otherwise you could add a true `merge` function that either merges two or many configs and piping global and local configs into that in the API